pkg/kubectl/cmd/util/factory_object_mapping.go

```diff
@@ -143,15 +148,26 @@ func (f *ring1Factory) UnstructuredClientForMapping(mapping *meta.RESTMapping) (
 	return restclient.RESTClientFor(cfg)
 }
 
-func (f *ring1Factory) Describer(mapping *meta.RESTMapping) (kubectl.Describer, error) {
+func printerForRing1Factory(f *ring1Factory, cmd *cobra.Command) (printers.ResourcePrinter, bool, error) {
+	mapper, typer := f.Object()
+	// TODO: used by the custom column implementation and the name implementation, break this dependency
+	decoders := []runtime.Decoder{f.clientAccessFactory.Decoder(true), unstructured.UnstructuredJSONScheme}
+	return PrinterForCommand(cmd, mapper, typer, decoders)
+}
+
+func (f *ring1Factory) PrinterForCommand(cmd *cobra.Command) (printers.ResourcePrinter, bool, error) {
+	return printerForRing1Factory(f, cmd)
```

Comment:
@smarterclayton @fabianofranz  I think this still breaks because you're accessing a peer method. If someone tries to override `ring1Factory.Object()` and then expects to be able to delegate for this method, then the wrong `f` will be used and the wrong `f.Object()` will be called.